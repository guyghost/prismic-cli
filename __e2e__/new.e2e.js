const path = require('path');
const fs = require('fs');
const { spawnSync } = require('child_process');
const {
  login,
  changeBase,
  deleteRepo,
  TMP_DIR,
  PRISMIC_BIN,
  rmdir,
  mkdir,
  genRepoName,
} = require('./utils');

describe('prismic new', () => {
  jest.setTimeout(300000);

  const dir = path.join(TMP_DIR, 'test-new');
  const repoName = genRepoName('cli-new-test');

  beforeAll(async () => {
    changeBase();
    login();
    await deleteRepo(repoName);
    return rmdir(dir, { recursive: true }).finally(() => {
      return mkdir(TMP_DIR, { recursive: true });
    });
  });


  it('should create a new project from a template', () => {
    const args = [
      'new',
      '--domain', repoName,
      '--folder', dir,
      '--template',
      'NodeJS',
      '--skip-install',
    ];
    const res = spawnSync(PRISMIC_BIN, args, { encoding: 'utf8', shell: true });
    const config = path.resolve(dir, 'prismic-configuration.js');
    expect(fs.existsSync(config)).toBeTruthy();
    // expect(res.stdout).toMatchSnapshot();
    expect(res.status).toBeFalsy();
  });
});
