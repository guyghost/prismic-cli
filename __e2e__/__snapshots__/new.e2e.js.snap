// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prismic new should create a new project from a template 1`] = `
"CAREFUL, your current base is: https://wroom.io

Let's get to it!
Initialize local project
Running npm install...
Your project is ready, to proceed:

Go to the project folder : cd /Users/marc/Projects/prismic/prismic-cli/__tmp__/test-new
To have hot reload: npm install -g nodemon
Start your project: nodemon app.js
Point your browser to: http://localhost:3000
"
`;
